### PR TITLE
Add + and -(Triangular, UniformScaling) methods

### PR DESCRIPTION
**Pull request by andreasnoack**
and move UniformScaling tests to separate file.

### CODE REVIEW

File: base/linalg/uniformscaling.jl [hunk 1]
```diff
@@ -24,12 +24,29 @@ one{T}(J::UniformScaling{T}) = one(UniformScaling{T})
...
+    UpperTriangular(Unew)
+end
+(+)(L::LowerTriangular, J::UniformScaling) = LowerTriangular(L.data + J)
+function (+)(L::LinAlg.UnitLowerTriangular, J::UniformScaling)
+    Lnew = copy(L.data)
+    for i = 1:size(Lnew, 1)
+        Lnew[i,i] = 2
+    end
+    LowerTriangular(Lnew)
+end
```
Comment (carnaval) [inline base/linalg/uniformscaling.jl]: can you define it for `{T<:Union{Lower,Upper}}(UL::T, ::UniformScaling)` and return `T(ULnew)` at the end ? to save the repetition

Comment (carnaval) [inline base/linalg/uniformscaling.jl]: nevermind, didn't read this well

Comment (carnaval) [inline base/linalg/uniformscaling.jl]: but I think you are assuming J.lambda == 1 here, and in the - case also

Comment (andreasnoack) [inline base/linalg/uniformscaling.jl]: Yes, that is not correct.

Comment (carnaval) [inline base/linalg/uniformscaling.jl]: does that makes me a linear algebra expert ? is there a club somewhere with cigars and whiskey ?

File: test/linalg/uniformscaling.jl [hunk 2]
```diff
@@ -0,0 +1,90 @@
+using Base.Test
```
Comment (tkelman) [inline test/linalg/uniformscaling.jl]: run `./julia contrib/add_license_to_files.jl`

File: base/linalg/uniformscaling.jl (hunk 1 above)
Comment (jiahao) [inline base/linalg/uniformscaling.jl]: I won't complain if cigars show up in Julia Central.

Comment (andreasnoack) [inline base/linalg/uniformscaling.jl]: @carnaval They are all over the world, but extreme exclusive. I bit like

https://www.bostonglobe.com/metro/2015/06/30/tom-and-gisele-looking-join-the-country-club/ButdqQMWmHLKcQ3QGCQGaL/story.html#

Just substitute "golf" with "numerical linear algebra" and you are close.

**Comment by kshyatt** [comment]
So these tests were not run [as we can see on Coveralls](https://codecov.io/github/JuliaLang/julia/base/linalg/uniformscaling.jl?ref=d88dd21e0d5722377bb4fe59e7330d35bae48081), and now that I am running them, they are failing.

File: base/linalg/uniformscaling.jl [hunk 3]
```diff
@@ -24,14 +24,63 @@ one{T}(J::UniformScaling{T}) = one(UniformScaling{T})
...
+        ULnew = similar(UL, promote_type(eltype(J), eltype(UL)))
+        n = size(ULnew, 1)
+        ULold = UL.data
+        for j = 1:n
+            ULnew[j,j] = J.λ - ifelse(isa(UL, UnitLowerTriangular), 1, ULold[j,j])
+            for i = j + 1:n
+                ULnew[i,j] = -ULold[i,j]
+            end
+        end
+        return LowerTriangular(ULNew)
```
Comment (kshyatt) [inline base/linalg/uniformscaling.jl]: `ULNew` is a variable that does not exist.

**Comment by kshyatt** [comment]
The only method which has passing tests is the one for `UpperTriangular`. The others all fail with a `BoundsError` in `setindex!` on the line to update the main diagonal.

File: base/linalg/uniformscaling.jl (hunk 3 above)
Comment (andreasnoack) [inline base/linalg/uniformscaling.jl]: I believe it does. It is defined in line 70, but it should probably be `similar(UL.data`. I'll look into that. The first problem is to figure how to modify the signature such that Julia considers a `XTriangularMatrix` more specific than an `AbstractMatrix`.

Comment (kshyatt) [inline base/linalg/uniformscaling.jl]: If it helps at all, I did open a PR #12297 which is probably an ugly fix.

**Pull request merged**